src/infrastructure/data_reference.py

```
@@ -0,0 +1,162 @@
+    def get_processed_path(self, source: str) -> Path:
+        if source not in self.data_sources:
+            raise ValueError(f"Unknown data source: {source}")
+        return self.data_sources[source]
+
+    def get_raw_path(self, source: str) -> Path:
+        if source not in self.raw_sources:
+            raise ValueError(f"No raw path configured for source: {source}")
+        return self.raw_sources[source]
+
+    def list_files(self, source: str, pattern: str = "*",
+                   processed: bool = True) -> List[Path]:
```

Ruff (0.12.2)

Boolean-typed positional argument in function definition

(FBT001)